### PR TITLE
End-to-end testing of python logging -> store ingestion

### PR DESCRIPTION
**Pull request by emilk**
Part of #1483

Try it with: `scripts/run_python_e2e_test.py` (must be in the correct venv).

This tests:
* Our Python API
* `LogMsg` encoding/decoding
* Arrow encoding/decoding
* TCP connection
* Data store ingestion

~This is not yet run on CI.~ I've also added it to CI (waiting to see how that turns out…)

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -216,6 +216,9 @@ jobs:
       - name: Run tests
         run: cd rerun_py/tests && pytest
 
+      - name: Run e2e test
+        run: scripts/run_python_e2e_test.py
```
Comment (jleibs) [inline .github/workflows/python.yml:220]: If we are running this as part of the wheel job, there's not a great reason to run:
```
subprocess.Popen(["just", "py-build", "--quiet"], env=build_env).wait()
```
below.  The wheel is installed into the local environment just a few lines up from here.

Comment (emilk) [reply .github/workflows/python.yml:220]: True, but when running it from the command line it's quite useful… I guess I could make it opt-out

Comment (emilk) [reply .github/workflows/python.yml:220]: Let's see how much time it adds to the CI

File: scripts/run_python_e2e_test.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    )
+    time.sleep(0.3)  # Wait for rerun server to start to remove a logged warning
+
+    python_process = subprocess.Popen([python_executable, example, "--connect", "--addr", f"127.0.0.1:{port}"])
+
+    print("Waiting for python process to finish…")
+    python_process.wait(timeout=30)
+
+    print("Waiting for rerun process to finish…")
+    rerun_process.wait(timeout=30)
```
Comment (jleibs) [inline scripts/run_python_e2e_test.py:73]: Have you tested that if something fails this "breaks" successfully? My expectation would be we potentially need to check the return code here?

Comment (emilk) [reply scripts/run_python_e2e_test.py]: Good catch - fixed!

File: .github/workflows/python.yml (hunk 1 above)
Comment (teh-cmc) [reply .github/workflows/python.yml:220]: > True, but when running it from the command line it's quite useful… I guess I could make it opt-out

I believe you can silently skip it if the `CI` env var is set.

**Review by teh-cmc — APPROVED** [review]

**Comment by emilk** [comment]
We finally got the e2e running when merged to `main`:


```
Run scripts/run_python_e2e_test.py
----------------------------------------------------------
Building rerun-sdk…
Traceback (most recent call last):
  File "scripts/run_python_e2e_test.py", line 79, in <module>
    main()
  File "scripts/run_python_e2e_test.py", line 28, in main
    subprocess.Popen(["just", "py-build", "--quiet"], env=build_env).wait()
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'just'
Error: Process completed with exit code 1.
```

**Pull request merged**